[ML] Changes DFA intro text

szabosteve opened the pull request:
## Summary

This PR updates the DFA intro on the overview page in the ML app.

![Screenshot 2022-02-01 at 10 31 47](https://user-images.githubusercontent.com/22324794/151943987-5e26e0bd-a151-4f29-9e31-7f7ca399c076.png)

### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)

lcawl commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/empty_prompt/empty_prompt.tsx:
```diff
@@ -63,7 +63,7 @@ export const AnalyticsEmptyPrompt: FC<Props> = ({ disabled, onCreateFirstJobClic
             <FormattedMessage
               id="xpack.ml.overview.analyticsList.emptyPromptText"
-              defaultMessage="Data frame analytics enables you to perform outlier detection, regression, or classification analysis and put the annotated data in a new index. The classification and regression trained models can also be used for inference in pipelines and aggregations."
+              defaultMessage="Perform outlier detection, regression, or classification analysis and put the annotated data in a new index. You can also use trained models for inference in pipelines and aggregations."
```
This is nice and short and snappy. I wonder if we need to make it clearer that we generate the trained models as part of this process. For example, switch from "use" to "create":
```suggestion
              defaultMessage="Perform outlier detection, regression, or classification analysis and put the annotated data in a new index. Create trained models to add inference in pipelines and aggregations."
```

lcawl commented:
I think we should also try to make this briefer too, though it's going to be tough to prune:
> defaultMessage="Data frame analytics requires specifically structured source data. Use {transforms} to create data frames before you create the jobs."

Also, IMO we should add a link to provide more info about exactly how the data must be structured (e.g. https://www.elastic.co/guide/en/machine-learning/master/ml-dfa-overview.html#prepare-transform-data?) Otherwise, we're sending folks to the transforms app without a lot of info about what they need to do there.

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

szabosteve commented:
@elasticmachine merge upstream

szabosteve commented:
@lcawl I've added a link to the "Prepare and transform" data section to the helper text and deleted `Elasticsearch` from the text to make it shorter.

peteharverson reviewed (approved):
LGTM

grabowskit commented:
I suggest we change “Perform outlier detection, regression, or classification analysis and put the annotated data in a new index. Create trained models to add inference in pipelines and aggregations.”  I don't think we need to mention the new index or inference in pipelines and aggregations.  

I like the idea of keeping the statement simple and to the point, like: "Build and train an outlier detection, regression, or classification machine learning models using data frame analysis." 

I would even change the note at the bottom to be more encouraging... "Before building a data frame analysis job you might want to use Transforms to construct a dataset by pivoting around an entity (user, host, customer, etc.) to analyze"

szabosteve commented:
@grabowskit Thanks for your comment. I changed the text to be in line with your suggestion. I also changed the note on the bottom, however, I haven't added the brackets with the examples to keep the text a bit more generic.

lcawl commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/empty_prompt/empty_prompt.tsx:
```diff
@@ -63,21 +63,34 @@ export const AnalyticsEmptyPrompt: FC<Props> = ({ disabled, onCreateFirstJobClic
             <FormattedMessage
               id="xpack.ml.overview.analyticsList.emptyPromptText"
-              defaultMessage="Data frame analytics enables you to perform outlier detection, regression, or classification analysis and put the annotated data in a new index. The classification and regression trained models can also be used for inference in pipelines and aggregations."
+              defaultMessage="Build and train outlier detection, regression, or classification machine learning models using data frame analytics."
```
Not mandatory, but IMO there's not a significant difference between "build" and "train" here, so I'd drop one:
```suggestion
              defaultMessage="Train outlier detection, regression, or classification machine learning models using data frame analytics."
```

lcawl commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/empty_prompt/empty_prompt.tsx:
```diff
@@ -63,21 +63,34 @@ export const AnalyticsEmptyPrompt: FC<Props> = ({ disabled, onCreateFirstJobClic
               <FormattedMessage
                 id="xpack.ml.overview.analyticsList.emptyPromptHelperText"
-                defaultMessage="Data frame analytics requires specifically structured source data. Use {transforms} to create data frames before you create the jobs."
+                defaultMessage="Before building a data frame analytics job, use {transforms} to construct an {sourcedata} to analyze."
```
Nit, but IMO "to analyze" is unnecessary here:
```suggestion
                defaultMessage="Before building a data frame analytics job, use {transforms} to construct an {sourcedata}."
```

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 124038` or prevent reminders by adding the `backport:skip` label.
